pkg/store/kotsstore/downstream_store.go

```diff
@@ -508,6 +524,51 @@ func getDownstreamVersionStatus(status types.DownstreamVersionStatus, hasError s
 	return s
 }
 
+func needsKotsUpgrade(app *kotsv1beta1.Application) bool {
+	if app == nil {
+		return false
+	}
+	if app.Spec.MinKotsVersion == "" && app.Spec.TargetKotsVersion == "" {
```

Comment:
this check won't be needed anymore when using `IsKotsVersionCompatibleWithApp` instead